src/check/parse/tokenize.zig

```diff
@@ -1016,23 +1022,21 @@ pub const Tokenizer = struct {
 
                 // Numbers starting with 0-9
                 '0'...'9' => {
-                    _ = self.cursor.chompNumber(b);
+                    const tag = self.cursor.chompNumber(b);
                     const len = self.cursor.pos - start;
-                    try self.output.pushToken(.Int, start, len);
+                    try self.output.pushToken(tag, start, len);
                 },
 
                 // Lowercase identifiers
                 'a'...'z' => {
-                    self.cursor.pos += 1;
-                    _ = self.cursor.chompIdentLower();
+                    const tag = self.cursor.chompIdentLower();
                     const len = self.cursor.pos - start;
-                    try self.output.pushToken(.LowerIdent, start, len);
+                    try self.output.pushToken(tag, start, len);
                 },
 
                 // Uppercase identifiers
                 'A'...'Z' => {
-                    self.cursor.pos += 1;
```

Comment:
Ditto

Reply:
I think it is necessary to get token offsets correct like above. That said, probably can instead let the ident functions know they have an extra character to care about.